ProcessLib/GroundwaterFlowProcess.h

```diff
@@ -208,6 +216,18 @@ class GroundwaterFlowProcess : public Process
         _local_to_global_index_map.reset(
             new AssemblerLib::LocalToGlobalIndexMap(_all_mesh_subsets));
 
+#ifdef USE_PETSC
+        DBUG("Allocate global matrix, vectors, and linear solver.");
+        MathLib::PETScMatrixOption mat_opt;
+        const MeshLib::NodePartitionedMesh &pmesh
+                    = static_cast<const MeshLib::NodePartitionedMesh&>(_mesh);
+        mat_opt.d_nz = pmesh.getMaximumNConnectedNodesToNode();
+        mat_opt.o_nz = mat_opt.d_nz;
+        _A.reset(_global_setup.createMatrix(_local_to_global_index_map->dofSizeGlobal(), mat_opt) );
+        _x.reset(_global_setup.createVector(_local_to_global_index_map->dofSizeGlobal()));
+        _rhs.reset(_global_setup.createVector(_local_to_global_index_map->dofSizeGlobal()));
+        _linearSolver.reset(new typename GlobalSetup::LinearSolver(*_A, "gw_"));
```

Comment:
Is the second argument to the LinearSolver ctor necessary?
If yes, please prepare a PR unifying existing LinearSolver ctor concepts.
Otherwise I strongly recommend to stick to the existing concept of LinearSolvers.